src/python/pants/backend/python/lint/bandit/rules_integration_test.py

```diff
@@ -31,7 +39,16 @@ def alias_groups(cls) -> BuildFileAliases:
 
   @classmethod
   def rules(cls):
-    return (*super().rules(), *bandit_rules(), RootRule(BanditTarget))
+    return (
+      *super().rules(),
+      *bandit_rules(),
+      download_pex_bin.download_pex_bin,
+      *pex.rules(),
+      *python_native_code.rules(),
+      *subprocess_environment.rules(),
+      RootRule(BanditTarget),
+      RootRule(download_pex_bin.DownloadedPexBin.Factory),
+    )
```

Comment:
Hm, this is a bit of a regression. The purpose of https://github.com/pantsbuild/pants/pull/9059 was to only need to have `(*super().rules(), *bandit_rules(), RootRule(BanditTarget))` in the test runner and nothing else.

Can you please check these things:
1) Can you remove `*pex.rules(), *python_native_code.rules(), *subprocess_environment.rules()` from here and see if the tests work?
2) Does this tool still work if you remove from `pants.ini` `pants.backend.python` and only have `pants.backend.python.lint.bandit` enabled?

If possible, it'd also be great to get the `DownloadPexBin.Factory` thing working so that you don't need to special-case it here. I know you were having some trouble with that. What issue were you running into?

Reply:
> I know you were having some trouble with that. What issue were you running into?

I don't recall this issue?

> The purpose of #9059 was to only need to have (*super().rules(), *bandit_rules(), RootRule(BanditTarget))

This PR was from before then, so I suspect this change will work.

Reply:
Ok, removing the same from the python awslambda test didn't work. I don't know if this was really "special-cased" as you describe -- I recall @gshuflin had this issue yesterday where we needed to inject `PythonSetup` in this exact way. I in fact used this PR to demonstrate how that was necessary. I'm not sure what kind of special-casing you're referring to with this comment and I am under the impression that injecting these as `RootRule`s is correct. As @gshuflin was also discussing with me yesterday, if you're concerned about the boilerplate, factoring this code into a common test base class seems appropriate.

Reply:
>  I am under the impression that injecting these as RootRules is correct

We do not want to register subsystems as `RootRule`s in tests, nor do we want to have to call `init_subsystem` in V2 tests. The `subystem_rule` declaration from production code should ideally work with the tests too. 

Note that these lint tests did not previously need to call `RootRule(PythonSetup)`, `RootRule(SourceRootConfig)`, etc. 

Originally, we actually did have to do this, but thanks to https://github.com/pantsbuild/pants/pull/8943 we no longer need to because we pass `OptionsBootstrapper` to the tests.

--

> I recall @gshuflin had this issue yesterday where we needed to inject PythonSetup in this exact way.

We figured out the issue yesterday and no longer need to pass `RootRule(PythonSetup)`. The issue yesterday was that we were passing `RootRule(BuildRoot)` when we shouldn't have, which meant then registering a bunch of rules that were unnecessary. See https://github.com/pantsbuild/pants/pull/9077/files#diff-dbe7c30c024e6830b7a7786c4521fbf2 for what we ended up registering.

> if you're concerned about the boilerplate, factoring this code into a common test base class seems appropriate.

That's one option, but even better would be being able to go back to the only boilerplate being the test having to register this:

```python
def rules():
  return (*super().rules(), *bandit_rules(), RootRule(BanditTarget))
```

The first win would be to try this:
> Can you remove *pex.rules(), *python_native_code.rules(), *subprocess_environment.rules() from here and see if the tests work?

The next win would be 

> It'd also be great to get the DownloadPexBin.Factory thing working so that you don't need to special-case it here

Per this, `Ok, removing the same from the python awslambda test didn't work.`, it sounds like that second part is a little harder to do though. What is the issue you run into with that second change?